### PR TITLE
Alerting: Use folders' RBAC permission to control rules actions

### PR DESCRIPTION
**Pull request by konrad147**
**What this PR does / why we need it**:
This PR makes use of the new `accessControl` property in the `folders` API to improve the handling of folder-scoped permissions

**Which issue(s) this PR fixes**:
Fixes https://github.com/grafana/alerting-squad/issues/297

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/core/services/backend_srv.ts [hunk 1]
```diff
@@ -433,8 +433,13 @@ export class BackendSrv implements BackendService {
     return this.get<DashboardDTO>(`/api/public/dashboards/${uid}`);
   }
 
-  getFolderByUid(uid: string) {
-    return this.get<FolderDTO>(`/api/folders/${uid}`);
+  getFolderByUid(uid: string, options: { withAccessControl?: boolean } = {}) {
```
Comment (gillesdemey) [inline public/app/core/services/backend_srv.ts:436]: nit: not a big fan of these inline type definitions, can we move this to a separate interface?

File: public/app/core/services/context_srv.ts [hunk 2]
```diff
@@ -174,7 +174,7 @@ export class ContextSrv {
   }
 
   hasAccessInMetadata(action: string, object: WithAccessControlMetadata, fallBack: boolean) {
-    if (!config.rbacEnabled) {
```
Comment (gillesdemey) [inline public/app/core/services/context_srv.ts:177]: Q: What's the difference between `rbacEnabled` and `accessControlEnabled()`?

File: public/app/features/alerting/unified/state/actions.ts [hunk 3]
```diff
@@ -568,7 +569,7 @@ export const deleteTemplateAction = (templateName: string, alertManagerSourceNam
 
 export const fetchFolderAction = createAsyncThunk(
   'unifiedalerting/fetchFolder',
-  (uid: string): Promise<FolderDTO> => withSerializedError((getBackendSrv() as any).getFolderByUid(uid))
+  (uid: string): Promise<FolderDTO> => withSerializedError(backendSrv.getFolderByUid(uid, { withAccessControl: true }))
```
Comment (gillesdemey) [inline public/app/features/alerting/unified/state/actions.ts:572]: 👌 no more any typing

**Review by gillesdemey — APPROVED** [review]
Some nits and questions, but no blockers

File: public/app/core/services/context_srv.ts (hunk 2 above)
Comment (konrad147) [reply public/app/core/services/context_srv.ts:177]: There should be no difference. `accessControlEnabled` wraps `rbacEnabled`. Other methods use the method so I unified

**Review by peterholmberg — APPROVED** [review]

**Review by IevaVasiljeva — COMMENTED** [review]
All looks good from RBAC perspective 👌

**Pull request merged**